packages/nimble-components/src/table/specs/table-column-width-hld.md

```diff
@@ -0,0 +1,185 @@
+-   If a user drags a divider between two columns to the right, then the column on the left will grow larger, and the column on the right will grow smaller by the same pixel amount. Sub-behaviors to this are:
+    -   If a shrinking column has reached its minimum pixel size or is not resizable, then the next column in the direction of the sizing action will be affected up to the final column in a given direction.
+    -   A sizing action to the left will ultimately stop having an effect when the left-most column reaches its minimum size.
+    -   A sizing action to the right that would ultimately result in the final right column reaching its minimum size (all columns still within current table width) would then begin to push columns out of the table width resulting in a horizontal scrollbar.
+-   Columns can be configured to not allow a user to interactively size them
+    -   The implicit behavior present based on the behaviors described above, is that in a sizing action that cascades to a column configured to not be resized is that the column won't be resized towards a minimum size, and the cascade will effectively "skip" this column.
+
+### Out of Scope
+
+There are some column sizing behaviors that we will ultimately expect to support, but the APIs presented here are not meant to address:
+
+-   Auto-resizing: We will not describe how we intend to support the use-case of having a column auto-size to its contents
+-   Different interactive sizing modes: While the APIs described in this HLD do not inherently prescribe to a particular interactive sizing behavior, it's worth saying that in order to support multiple sizing modes, there will likely be additional APIs required that this HLD does not address.
+-   Mechanisms related to accessibility-centric interactive column sizing (if there are such mechanisms). One possible example is allowing a user to size a column by way of the keyboard, instead of using a mouse. Ultimately, such a scenario is not in conflict with the API presented here, nor the mouse-based approach we know we will require, and can thus be handled separately, if ever.
+
+### API
+
+To accomodate the various sizing modes of a column, in addition to the other necessary behaviors, we can add the following properties to `TableColumn`:
+
+```
+abstract class TableColumn {
+    /*
+     * @internal
+     */
+    // when set 'currentFractionalWidth' will be ignored
+    @observable
+    public currentPixelWidth: number | null = null;
+
+    /*
+     * @internal
+     */
+    @observable
+    public currentFractionalWidth = 1;
+
+    /*
+     * @internal
+     */
+    @observable
+    public columnMinPixelWidth = 88; // the minimum size in pixels according to the design doc
```

Comment:
We can potentially hash out exact naming when we get to the implementation, but it seems inconsistent to have `columnMinPixelWidth` and `columnDisableResize` to both start with the term "column" when none of the other properties do. Unless there is something that makes this properties potentially ambiguous in their meaning, I'd vote for just calling them `minPixelWidth` and `disableResize`.

Reply:
We converged on the naming of the properties in this [comment thread](https://github.com/ni/nimble/pull/1002#discussion_r1099354412). The quick summary is that those properties need a "uniquified" name as the more succinct names are being reserved for the mixin API.

Reply:
Related to this, had two thoughts:
- It would be nice if the responsibilities were clearer for who owned what values. ie the plugin author write to and always owned some property and the table "owned" a separate property where it was writing updated values.
- If we can use a prefix that made it clearer at runtime for a JS developer using devtools to understand the difference of the properties. ie column instances in devtools using the mixin would have `minPixelWdith` and `columnMinPixelWidth` and it's hard to know the difference as a client between them (unless I look at the TS definitions and see that the column prefixes are internal) 

Talked to Meyers and proposed the following:
```
currentPixelWidth
currentFractionalWidth

internalPixelWidth
internalFractionalWidth
internalMinPixelWidth
internalDisableResize
```

The table would write to the "current" prefix properties and the plugin author would write to the "internal" prefix properties.